app/controllers/product_controller.rb

```diff
@@ -1,5 +1,13 @@
 class ProductController < ApplicationController
   def index
     @products = Category.find(params[:id]).products
+    @amount_cart = amount_sum
+    @shop_id = Category.find(params[:id]).shop_id
+  end
+
+  private
+
+  def amount_sum
+    CartItem.where(cart_id: @cart.id).sum(&:amount)
```

Comment:
@amount_sum = CartItem.where(cart_id: @cart.id).sum(&:amount)

and
before_action :amount_sum, only :index